src/sources/timer/timerfd.rs

```diff
@@ -33,7 +33,15 @@ impl TimerScheduler {
 
     pub fn reschedule(&mut self, new_deadline: Instant) {
         let now = Instant::now();
-        let time = TimeSpec::from_duration(new_deadline.duration_since(now));
+
+        // We should handle the case when duration is zero. Since timerfd can't do that we pass the
+        // timer resolution, which is 1ns triggering the timer right away.
+        let time = TimeSpec::from_duration(
+            new_deadline
+                .checked_duration_since(now)
+                .unwrap_or_else(|| Duration::from_nanos(1)),
+        );
```

Comment:
What happens in the extremely unlikely case where `now == new_deadline` ?

Reply:
Yeah, you're right, we should use fallback duration for this as well.